### PR TITLE
make dependency on typify-macro optional

### PR DESCRIPTION
**Pull request by jayvdb**
c.f. https://github.com/oxidecomputer/progenitor/pull/345

happy to create a feature "macro" or "macros" if you prefer.

### CODE REVIEW

**Comment by ahl** [comment]
What's the benefit of this? Are you seeing a significantly faster compilation of progenitor without the typify-macro dependency?

**Comment by jayvdb** [comment]
Main benefit is reduced SBOM/SOUP(Software of Unknown Provenance).  In automotive/medical/etc, every dependency needs to be recorded,audited,etc, so removing unused `typify-macro` and `serde_tokenstream` is a win.

File: typify/Cargo.toml [hunk 1]
```diff
@@ -10,8 +10,11 @@ readme = "../README.md"
 keywords = ["json", "schema", "proc_macro"]
 categories = ["api-bindings", "compilers"]
 
+[features]
+default = ["typify-macro"]
```
Comment (ahl) [inline typify/Cargo.toml:14]: let's call it "macro"

**Review by ahl — CHANGES_REQUESTED** [review]
ok

**Review by ahl — APPROVED** [review]

**Pull request merged**